### PR TITLE
Decoders for curve pools deposits/withdrawals

### PR DESCRIPTION
**Pull request by yabirgb**
This PR introduces a new type of "enrichers" for eth send in the transaction. This happens in some pools that accept ETH as deposit

image

### CODE REVIEW

File: rotkehlchen/chain/ethereum/decoding/decoder.py [hunk 1]
```diff
@@ -135,23 +138,26 @@ def _recursively_initialize_decoders(
                     address_results.update(self.decoders[class_name].addresses_to_decoders())
                     rules_results.extend(self.decoders[class_name].decoding_rules())
                     enricher_results.extend(self.decoders[class_name].enricher_rules())
+                    ethereum_send_result.extend(self.decoders[class_name].ethereum_send_rules())
 
-                recursive_addrs, recursive_rules, recurisve_enricher_results = self._recursively_initialize_decoders(full_name)  # noqa: E501
+                recursive_addrs, recursive_rules, recurisve_enricher_results, recursive_ethereum_send_result = self._recursively_initialize_decoders(full_name)  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/decoding/decoder.py:143]:
```suggestion
                recursive_addrs, recursive_rules, recursive_enricher_results, recursive_ethereum_send_result = self._recursively_initialize_decoders(full_name)  # noqa: E501
```

File: rotkehlchen/chain/ethereum/decoding/decoder.py [hunk 2]
```diff
@@ -637,6 +645,15 @@ def _enrich_protocol_tranfers(  # pylint: disable=no-self-use
             if transfer_enriched:
                 break
 
+    def _enrich_eth_sent(self, tx: EthereumTransaction, event: HistoryBaseEntry) -> None:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/decoding/decoder.py:648]: Do we really need a different enricher for ETH sent? Shouldn't we just be using the transfer enricher for all transfers? Including tokens and ETH?

**Review by LefterisJP — COMMENTED** [review]
A very quick question from a quick look. Still need to look it up more.

File: rotkehlchen/chain/ethereum/decoding/decoder.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/decoding/decoder.py:648]: Do you mean avoid creating a new type of enricher and just use `_enrich_protocol_tranfers` instead of `_enrich_eth_sent` and have proper checks for the asset in the `_enrich_protocol_tranfers` implementation of the modules that need it?

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/decoding/decoder.py:648]: I mean we should have an enricher of all types of transfers. Be it token or ETH transfer. Should not differentiate between the 2.

File: rotkehlchen/chain/ethereum/decoding/decoder.py [hunk 3]
```diff
@@ -461,7 +461,7 @@ def _maybe_decode_simple_transactions(
         if amount == ZERO:
             return events
 
-        events.append(HistoryBaseEntry(
+        eth_sent_event = HistoryBaseEntry(
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/decoding/decoder.py:464]: why this change?

File: rotkehlchen/tests/unit/decoders/test_curve.py [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+    EthereumInternalTransaction,
+    EthereumTransaction,
+    Location,
+    Timestamp,
+    deserialize_evm_tx_hash,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.hexbytes import hexstring_to_bytes
+
+COUNTERPARTY = 'curve'
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/test_curve.py:27]: import `CPT_CURVE` instead please

**Review by LefterisJP — COMMENTED** [review]
Needs rebase to fix the frontend conflict. Should be easy. Left question and small comment.

File: rotkehlchen/chain/ethereum/decoding/decoder.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/decoding/decoder.py]: Because I reverted the changes I made here with the eth transfers but forgot to undo this one. Should be fixed now

Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/decoding/decoder.py]: I see. Expected something like that but was not sure.

**Review by LefterisJP — COMMENTED** [review]
Thank you for the changes. Seems like a test is failing now that was not before and is only failing in your branch:

`rotkehlchen/tests/api/test_ens.py::test_reverse_ens`

Will need it figured out

**Comment by yabirgb** [comment]
> Thank you for the changes. Seems like a test is failing now that was not before and is only failing in your branch:
> 
> `rotkehlchen/tests/api/test_ens.py::test_reverse_ens`
> 
> Will need it figured out

It failed for me once but tried it several times and runs without issues

**Comment by LefterisJP** [comment]
> > Thank you for the changes. Seems like a test is failing now that was not before and is only failing in your branch:
> > `rotkehlchen/tests/api/test_ens.py::test_reverse_ens`
> > Will need it figured out
> 
> It failed for me once but tried it several times and runs without issues

That's not fine though. But yes then maybe not related to the PR. We can't have a test being flaky on something that's not a remote error.

The error is here for posterity:

```
        timestamps_before_request = [mapping.last_update for mapping in db.get_reverse_ens(all_addrs)]
        requests.post(
            api_url_for(
                rotkehlchen_api_server,
                "reverseensresource",
                ethereum_addresses=all_addrs,
                force_update=True,
            ),
        )
        timestamps_after_request = [mapping.last_update for mapping in db.get_reverse_ens(all_addrs)]
>       assert all(map(lambda t_pair: t_pair[0] < t_pair[1], zip(timestamps_before_request, timestamps_after_request)))  # noqa: E501
E       assert False
E        +  where False = all(<map object at 0x7f717845ea30>)
E        +    where <map object at 0x7f717845ea30> = map(<function test_reverse_ens.<locals>.<lambda> at 0x7f7193ebef70>, <zip object at 0x7f7190682380>)
E        +      where <zip object at 0x7f7190682380> = zip([1651611737, 1651611738, 1651611737, 1651611738], [1651611738, 1651611738, 1651611738, 1651611738])
```

**Review by LefterisJP — APPROVED** [review]
LGTM -- the test let's handle elsewhere
